### PR TITLE
Refactoring ResultFileHierarchy

### LINKED ISSUE #1031: Refactor `ResultFileHierarchy`

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #1031 

Changes in `ResultFileHierarchy`:
- Stripping unused fields
- Defining paths as java `Path`
- Removing unused paths in output directory, such as `kpi` or `graphs`
- Moving initialization to `apply` method
- Directories are now always created (removing unused parameter that toggled this)

### CODE REVIEW

File: src/test/scala/edu/ie3/simona/io/file/ResultFileHierarchySpec.scala [hunk 1]
```diff
@@ -106,37 +83,22 @@ class ResultFileHierarchySpec
...
     "should add a timestamp automatically on instantiation" in {} // todo
 
   }
 
-  private def relativizePath(fullPath: String): Path = {
-    new File(new File("").getAbsolutePath).toPath
-      .relativize(new File(fullPath).toPath)
-  }
+  private def relativizePath(fullPath: Path): Path =
+    new File(new File("").getAbsolutePath).toPath.relativize(fullPath)
```
Comment (staudtMarius) [inline src/test/scala/edu/ie3/simona/io/file/ResultFileHierarchySpec.scala:101]: I think this method can be removed, since it is not used anywhere.

**Review by staudtMarius — CHANGES_REQUESTED** [review]
Looks good to me. I just found two parts that coulb be improved. 

The argument `createDirs` in `SetupHelper.buildResultFileHierarchy` could be removed.

**Review by danielfeismann — APPROVED** [review]
LGTM!

File: src/test/scala/edu/ie3/simona/io/file/ResultFileHierarchySpec.scala (hunk 1 above)
Comment (sebastian-peter) [reply src/test/scala/edu/ie3/simona/io/file/ResultFileHierarchySpec.scala:101]: Good catch!

**Review by danielfeismann — APPROVED** [review]
LGTM

**Pull request merged**